### ISSUE TITLE
Allow deletion of unused smart clip data

### ISSUE THREAD

**Issue by adrien1018** [audacity/audacity#4715]
### Your idea

Add a function to delete unused / hidden smart clip data from the selected clips in order to reduce the project file size.

One way to achieve this is by disassociating the clips from the original clip data, and creating new clips that only contain the trimmed sections of the original clips. It will be equivalent to repeat the following process on each of the selected clips:
1. Move the clip to a new track
2. Perform "Mix and Render" on that track while preserving the clip label
3. Move the rendered clip back to its original position (and delete the new track)

This will allow future compaction to clean up the stale data if no other clips are using it.

### Problem to be solved

When editing a long original audio file, it is common to first select some usable clips by splitting and trimming the audio. However, currently there are no straightforward way to permanently trim the audio, so the project will keep storing the entire original file even most of them are unnecessary.

### Additional context

@virgilON suggested a similar idea in #3963 and #4047, though for a different (and solved) issue.

**Comment by JohnColket** [comment]
>  currently there are no straightforward way to permanently trim the audio, so the project will keep storing the entire original file even if most of them are unnecessary.

Yes, but with 3.2.x you can select the entire project, copy, then paste to a new project, skipping the smart clip stuff.

**Comment by adrien1018** [comment]
> Yes, but with 3.2.x you can select the entire project, copy, then paste to a new project, skipping the smart clip stuff.

That's clever! This could be documented.
It would have a (very minor) side effect of adding prefixes to the clip name though.

**Comment by petersampsonaudacity** [comment]
> That's clever! This could be documented.

It is documented in the **Preferences** page for **Tracks Behaviors**
https://manual.audacityteam.org/man/tracks_behaviors_preferences.html
But you could argue that that is a little obscure.

It's definitely **_not_** documented on the page about clips:
https://manual.audacityteam.org/man/audacity_tracks_and_clips.html
I will try to find some time to fix that as that is a more obvious place to look.

**Comment by petersampsonaudacity** [comment]
> Add a function to delete unused / hidden smart clip data from the selected clips in order to reduce the project file size.

I like and support this idea.

**Comment by adrien1018** [comment]
I've added @JohnColket 's workaround to a note in [Audacity Support](https://support.audacityteam.org/basics/audacity-editing#resizing-and-trimming-clips). Not sure whether it is discoverable enough, though.

**Comment by petersampsonaudacity** [comment]
Thanks @adrien1018  for the note on Audacity Support  :-)

I have now added this section (with @JohnColket 's workaround: 
https://alphamanual.audacityteam.org/man/Audacity_Projects#Permanently_delete_trimmed_audio

to this page:  
https://alphamanual.audacityteam.org/man/Audacity_Projects

I have added a P1 ednote as a flag to remove/modify the workaround if/when this logged issue gets fixed.

**Comment by SteveDaulton** [comment]
15 months later.

* You know that end users want this feature.
* You know that the absence of this feature is causing serious issues for many users.

Why are you so reluctant to listen to your end users?

**Comment by petersampsonaudacity** [comment]
See new user complaint:  
**Audacity creates insanely large file sizes** #7103

**Comment by FluffyRabb** [comment]
This would be amazing to have. Or even better an option on save project to "remove unused audio data". Or a preference to disable smart clip behaviour completely. Anything is reduce the project size without having to do a workaround.

**Comment by dascritch** [comment]
Why this problem is tagged as “feature” ? It is clearly a “regression” !

This is really hurting small benevolent podcast/radio operations that routinely does multi-track recording for broadcast as hard disk space is not completely free. A 4 track interview in mono, 1h30mn length actually weights for me 12Gb AND cause Audacity to lag terribly during editing, causing discomfort. Because the wal-based DB goes really encumbered with useless data, missing optimizations.

Doing the stupid ^C ^V workaround help to reduce it down from 12Gb to 4.8 Gb, but it sincerely should be done in the interface, as a “finalize project file”, not via a so space and time consuming activity.

We really urgently need a "clean up" feature for ready-to-broadcast and archival purposes. And as Undo/Redo is disabled when we re-open the project, it shows that all those data are completely unused/unreachable.

**Comment by dascritch** [comment]
Oh, BTW, "smart clip" feature is a pain in the ass when you try to select the edge of the audio clip. May be you should allow to deactivate it from the usual edition interface.
